classes/amity.py

```diff
@@ -1,45 +1,515 @@
+                livingspace_with_space.append(livingspace)
+        if livingspace_with_space:
+            random_livingspace = choice(livingspace_with_space)
+            random_livingspace.occupants.append(new_person.person_name)
+            new_person.living_space = random_livingspace
+            return "Allocated livingspace: {} \n". \
+                format(random_livingspace.room_name)
+        else:
+            self.waiting_list["livingspace"].append(new_person)
+            self.waiting_list["office"] = list(set(
+             self.waiting_list["office"]))
+            return "No available livingspaces. " + \
+                "Added to the livingspaces waiting list\n"
+
+    def print_room(self, room_name):
+        """A method that prints room occupants in a room"""
+        found = False
+        room_obj = ""
+        for room in (self.offices + self.livingspaces):
+            if room_name == room.room_name:
+                found = True
+                room_obj = room
+        if found:
+            if len(room_obj.occupants) == 0:
+                return "The room is empty!\n"
+            else:
+                for occupant in room_obj.occupants:
+                    print(occupant)
+                return "Room occupants printed successfully!\n"
+        else:
+            return "The room does not exist!\n"
 
     def reallocate_person(self, person_id, room_name):
-        """A method that is used to reallocate a person"""
-        pass
+        """A method that reallocates a person from one room to another"""
+        room_name = room_name.lower()
+        person_id = person_id.upper()
+        if person_id in [person.person_id for person in
+           (self.fellows + self.staffs)]:
+            if room_name in [room.room_name for room in
+               (self.offices + self.livingspaces)]:
+                if person_id[:3] == "FOO":
+                    if room_name in [room.room_name for room in self.offices]:
+                        return self.reallocate_fellow_office(person_id,
+                                                             room_name)
+                    else:
+                        return self.reallocate_fellow_livingspace(person_id,
+                                                                  room_name)
+                else:
+                    if room_name in [room.room_name for room in self.offices]:
+                        return self.reallocate_staff(person_id, room_name)
+                    else:
+                        return "Staff cannot be accomodated!"
 
-    def loads_people(self, file_name):
-        """A method that adds people from a text file"""
-        pass
+            else:
+                return "The room does not exist!"
+        else:
+            return "The person identifier(id) used does not exist!"
 
-    def print_allocated(self, file_name="none"):
+    def reallocate_fellow_office(self, person_id, room_name):
```

Comment:
look into building more reusable functions. The reallocation logic is the same for staff and fellows to either offices or living spaces. Rather than having 3 separate functions, you can have one that handles it all.